pkg/orchestrator/scheduler/batch_service_job.go

```diff
@@ -162,8 +187,23 @@ func (b *BatchServiceJobScheduler) placeExecs(ctx context.Context, execs execSet
 	return nil
 }
 
+func (b *BatchServiceJobScheduler) handleRetry(plan *models.Plan, job *models.Job, failures int) {
+	// delay = RetryDelay * RetryDelayGrowthFactor ^ failures
+	// ...but never more than MaximumRetryDelay
```

Comment:
I think this logic should fall under the `retryStrategy` where instead of returning true of false, it should return a response that says do not retry or do retry with a backoff value. This decouples the calculation of the backoff window from the scheduler into the retry strategy

Reply:
I've made a new ticket for this as it'll require a bit of rearrangement!